### ISSUE TITLE
960, 961 & 9620 & 9624 have been assigned to GS1 UK for GTIN-8 allocation

### ISSUE THREAD

**Issue by stereobooster** [jodal/biip#162]
As far as I can see from the source code it doesn't take into account this edge case

> 960, 961 & 9620 & 9624 have been assigned to GS1 UK for GTIN-8 allocation. 9625 & 9626 have been assigned to GS1 Poland for GTIN-8 allocation.

From https://www.gs1.org/standards/id-keys/company-prefix

**Comment by jodal** [comment]
You're absolutely right. The current implementation only loops through data similar to the table at https://www.gs1.org/standards/id-keys/company-prefix and returns the first match. To support these four edge cases, I think it needs to be rewritten to compare with the longest possible prefixes first.

I'm curious, what's your use case for knowing that these are not only "for GTIN-8" but also e.g. "assigned by GS1 UK"?

**Comment by stereobooster** [comment]
I don't have specific use case in my mind. I read through code-base and noticed it...

**Comment by jodal** [comment]
This has finally been fixed and will be part of the upcoming v4.0 release.

**Comment by jodal** [comment]
This has been released in [Biip 4.0.0](https://github.com/jodal/biip/releases/tag/v4.0.0).
